scenario_player/runner.py

```diff
@@ -2,46 +2,259 @@
+    assert contracts, msg
+
+    token_network_address = contracts["contracts"][CONTRACT_TOKEN_NETWORK_REGISTRY]["address"]
+
+    token_network_proxy = proxy_manager.token_network_registry(
+        TokenNetworkRegistryAddress(to_canonical_address(token_network_address)), "latest"
+    )
+    return token_network_proxy
+
+
+def determine_run_number(scenario_dir: Path) -> int:
+    """ Determine the current run number.
+
+    We check for a run number file, and use any number that is logged there
+    after incrementing it.
+    """
+    # TODO: Use advisory file locks to avoid concurrent read/writes on the
+    # run_number_file. Otherwise this can lead to very hard to debug races.
+    run_number_file = scenario_dir.joinpath(RUN_NUMBER_FILENAME)
+
+    if run_number_file.exists():
+        run_number = int(run_number_file.read_text()) + 1
+    else:
+        run_number = 0
+
+    run_number_file.write_text(str(run_number))
+
+    return run_number
+
+
+def make_session(auth: str, settings: SettingsConfig) -> Session:
+    session = Session()
+    if auth:
+        session.auth = cast(Tuple[str, str], tuple(auth.split(":")))
+    session.mount("http", TimeOutHTTPAdapter(timeout=settings.timeout))
+    session.mount("https", TimeOutHTTPAdapter(timeout=settings.timeout))
+
+    return session
+
+
+def wait_for_token_network_discovery(
+    node_endpoint: str, settings: SettingsConfig, session: Session
+) -> ChecksumAddress:
+    """Check for token network discovery with the given `node`.
+
+    By default exit the wait if the token has not been discovered after `n` seconds,
+    where `n` is the value of :attr:`.timeout`.
+
+    :raises TokenNetworkDiscoveryTimeout:
+        If we waited a set time for the token network to be discovered, but it wasn't.
+    """
+    started = time.monotonic()
+    elapsed = 0.0
+    while elapsed < settings.timeout:
+        try:
+            resp = session.get(node_endpoint)
+            resp.raise_for_status()
+
+        except HTTPError as e:
+            # We explicitly handle 404 Not Found responses only - anything else is none
+            # of our business.
+            if e.response.status_code != 404:
+                raise
+
+            # Wait before continuing, no sense in spamming the node.
+            gevent.sleep(1)
+
+            # Update our elapsed time tracker.
+            elapsed = time.monotonic() - started
+            continue
+
+        else:
+            # The node appears to have discovered our token network.
+            data = resp.json()
+
+            if not is_checksum_address(data):
+                # Something's amiss about this response. Notify a human.
+                raise TypeError(f"Unexpected response type from API: {data!r}")
+
+            return ChecksumAddress(data)
+
+    # We could not assert that our token network was registered within an
+    # acceptable time frame.
+    raise TokenNetworkDiscoveryTimeout
+
+
+def maybe_create_token_network(
```

Comment:
This corresponds to https://github.com/raiden-network/scenario-player/blob/edd041e816776c141a1182f089ab6bb75a976e91/scenario_player/runner.py#L264